File: Algorithm/5/removeElements.cpp
#include <iostream>

using namespace std;

/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode(int x) : val(x), next(NULL) {}
 * };
 */
class Solution {
public:
    ListNode* removeElements(ListNode* head, int val) {
        // while(head != NULL && head->val == val){
        //     ListNode* delNode = head;
        //     head = delNode->next;
        //     delete delNode;
        // }
        //
        // if(head == NULL)
        //     return NULL;
        // ListNode* cur = head;
        // while(cur->next != NULL ){
        //     if(cur->next->val == val){
        //         //É¾³ıcur->next
        //         ListNode* delNode = cur->next;
        //         cur->next = delNode->next;
        //         delete delNode;
        //         //delNode->next = NULL;
        //     }
        //     else
        //         cur = cur->next;
        // }
        // return head;

        ListNode* dummyHead = new ListNode(0);
        dummyHead->next = head;

        ListNode* cur = dummyHead;
        while(cur->next != NULL ){
            if(cur->next->val == val){
                //É¾³ıcur->next
                ListNode* delNode = cur->next;
                cur->next = delNode->next;
                delete delNode;
                //delNode->next = NULL;
            }
            else
                cur = cur->next;
        }

        ListNode* retNode = dummyHead->next;
        delete dummyHead;
        return retNode;
    }
};

int main() {

    return 0;
}

File: Algorithm/5/removeElements.py
# Definition for singly-linked list.
# class ListNode(object):
#     def __init__(self, x):
#         self.val = x
#         self.next = None

class Solution(object):
    def removeElements(self, head, val):
        """
        :type head: ListNode
        :type val: int
        :rtype: ListNode
        """
        # if head == None:
        #     return None
        # pre = ListNode(-1)
        # cur = head
        # pre.next = cur
        # while head != None and head.val == val:
        #     head = head.next
        #
        # while cur != None:
        #     if cur.val == val:
        #         pre.next = cur.next
        #         cur = cur.next
        #     else:
        #         pre = cur
        #         cur = cur.next
        # return head

        dummyHead = ListNode(-1)
        dummyHead.next = head
        cur = dummyHead;
        while cur.next != None:
            if cur.next.val == val:
                delNode = cur.next
                cur.next = delNode.next
                delNode.next = None
            else:
                cur = cur.next
        retNode = dummyHead.next
        return retNode
